### PR TITLE
Change `energy_range` to `energy_bounds` in `SpectralModel.plot()`

### PR DESCRIPTION
**Pull request by fabiopintore**
This PR changes the definition name of the `energy_range` parameter in `SpectralModel.plot()` and `SpectralModel.plot_error()`. This is replaced by `energy_bounds`. 
It solves an issue raised in #3078.

### CODE REVIEW

**Comment by adonath** [comment]
Thanks @fabiopintore, please adapt the remaining cases, where `energy_range` is used as variable name like:

```
examples/models/spectral/plot_constant_spectral.py
21:energy_range = [0.1, 100] * u.TeV
23:model.plot(energy_range)

examples/models/spectral/plot_exp_cutoff_powerlaw.py
25:energy_range = [0.1, 100] * u.TeV
32:model.plot(energy_range)

examples/models/spectral/plot_compound.py
26:energy_range = [0.1, 100] * u.TeV
34:model.plot(energy_range)

examples/models/spectral/plot_super_exp_cutoff_powerlaw_4fgl.py
33:energy_range = [0.1, 100] * u.TeV
41:model.plot(energy_range)

examples/models/spectral/plot_broken_powerlaw.py
27:energy_range = [0.1, 100] * u.TeV
31:model.plot(energy_range)

examples/models/spectral/plot_super_exp_cutoff_powerlaw_3fgl.py
31:energy_range = [0.1, 100] * u.TeV
39:model.plot(energy_range)

examples/models/spectral/plot_template_spectral.py
39:energy_range = [0.1, 1] * u.TeV
43:template.plot(energy_range)

examples/models/spectral/plot_powerlaw2.py
27:energy_range = [0.1, 100] * u.TeV
31:model.plot(energy_range)

examples/models/spectral/plot_absorbed.py
45:energy_range = [0.08, 3] * u.TeV
46:opts = dict(energy_range=energy_range, energy_unit="TeV", flux_unit="")
52:plt.xlim(energy_range.value)
71:energy_range = [0.1, 100] * u.TeV
73:model.plot(energy_range)

examples/models/spectral/plot_piecewise_norm_spectral.py
25:energy_range = [0.1, 100] * u.TeV
29:model.plot(energy_range, flux_unit="")

examples/models/spectral/plot_logparabola.py
36:energy_range = [0.1, 100] * u.TeV
40:model.plot(energy_range)

examples/models/spectral/plot_naima.py
48:    "energy_range": [10 * u.GeV, 80 * u.TeV],

examples/models/spectral/plot_powerlaw.py
25:energy_range = [0.1, 100] * u.TeV
29:model.plot(energy_range)

examples/models/spectral/plot_exp_cutoff_powerlaw_3fgl.py
25:energy_range = [0.1, 100] * u.TeV
32:model.plot(energy_range)

examples/models/spectral/plot_gauss_spectral.py
25:energy_range = [0.1, 100] * u.TeV
27:model.plot(energy_range)

examples/models/spectral/plot_smooth_broken_powerlaw.py
24:energy_range = [0.1, 100] * u.TeV
33:model.plot(energy_range)
```

And
```
docs/tutorials/analysis/3D/analysis_mwl.ipynb
343:    "energy_range = [0.01, 120] * u.TeV\n",
345:    "ax = crab_spec.plot(energy_range=energy_range, energy_power=2, label=\"Model\")\n",
346:    "crab_spec.plot_error(ax=ax, energy_range=energy_range, energy_power=2)\n",

docs/tutorials/analysis/3D/mcmc_sampling.ipynb
362:    "            energy_range=(emin, emax),\n",
371:    "    energy_range=(emin, emax), energy_power=2, ax=ax, color=\"red\"\n",

docs/tutorials/analysis/3D/analysis_3d.ipynb
677:    "    energy_range = [0.3, 10] * u.TeV\n",
679:    "        energy_range=energy_range, energy_power=2, label=label, color=color\n",
681:    "    spec.plot_error(energy_range=energy_range, energy_power=2, color=color)"

docs/tutorials/analysis/time/pulsar_analysis.ipynb
511:    "    energy_range=(emin_fit, emax_fit),\n",

docs/tutorials/analysis/1D/sed_fitting.ipynb
239:    "pwl.plot(energy_range=[1e-4, 1e2] * u.TeV, ax=ax, energy_power=2)\n",
240:    "pwl.plot_error(energy_range=[1e-4, 1e2] * u.TeV, ax=ax, energy_power=2)\n",
304:    "ecpl.plot(energy_range=[1e-4, 1e2] * u.TeV, ax=ax, energy_power=2)\n",
305:    "ecpl.plot_error(energy_range=[1e-4, 1e2] * u.TeV, ax=ax, energy_power=2)\n",
348:    "log_parabola.plot(energy_range=[1e-4, 1e2] * u.TeV, ax=ax, energy_power=2)\n",
350:    "    energy_range=[1e-4, 1e2] * u.TeV, ax=ax, energy_power=2\n",

docs/tutorials/analysis/1D/spectral_analysis.ipynb
665:    "    \"energy_range\": [0.1, 30] * u.TeV,\n",

docs/tutorials/starting/overview.ipynb
601:    "    energy_range=[10, 2000] * u.GeV, energy_power=0\n",
722:    "ax = crab_3fhl_spec.plot(energy_range=[10, 2000] * u.GeV, energy_power=2)\n",

docs/tutorials/starting/analysis_2.ipynb
528:    "energy_range = [1, 10] * u.TeV\n",
529:    "spec.plot(energy_range=energy_range, energy_power=2)\n",
530:    "ax = spec.plot_error(energy_range=energy_range, energy_power=2)"
570:    "ax = spec.plot_error(energy_range=energy_range, energy_power=2)\n",

docs/tutorials/api/fitting.ipynb
327:    "energy_range = [1, 10] * u.TeV\n",
328:    "crab_spectrum.plot(energy_range=energy_range, energy_power=2)\n",
329:    "ax = crab_spectrum.plot_error(energy_range=energy_range, energy_power=2)"

docs/tutorials/api/models.ipynb
232:    "pwl.plot(energy_range=[1, 100] * u.TeV)"
332:    "template.plot(energy_range=[0.2, 50] * u.TeV, label=\"template model\")\n",
335:    "    energy_range=[0.2, 50] * u.TeV, label=\"normed_template model\"\n",
774:    "model.spectral_model.plot(energy_range=[1, 10] * u.TeV);"
1128:    "my_custom_model.plot(energy_range=[1, 10] * u.TeV)"

docs/tutorials/api/catalog.ipynb
482:    "energy_range = (100 * u.MeV, 100 * u.GeV)\n",
484:    "model.spectral_model.plot(energy_range, **opts);\n",
485:    "model.spectral_model.plot_error(energy_range, **opts);"

docs/tutorials/api/astro_dark_matter.ipynb
205:    "            energy_range=[mDM / 100, mDM],\n",

docs/tutorials/data/fermi_lat.ipynb
427:    "energy_range = [50, 2000] * u.GeV\n",
428:    "diffuse_iso.spectral_model.plot(energy_range, flux_unit=\"1 / (cm2 MeV s)\");"

```

File: gammapy/modeling/models/spectral.py [hunk 1]
```diff
@@ -289,15 +289,15 @@ def plot(
...
+            ax = pwl.plot(energy_bounds=(0.1, 100) * u.TeV)
             ax.set_yscale('linear')
 
         Parameters
         ----------
         ax : `~matplotlib.axes.Axes`, optional
             Axis
-        energy_range : `~astropy.units.Quantity`
-            Plot range
+        energy_bounds : `~astropy.units.Quantity`
```
Comment (adonath) [inline gammapy/modeling/models/spectral.py:299]: Mention `energy_bounds` is forward to `MapAxis.from_energy_bounds()` here.

Comment (fabiopintore) [reply gammapy/modeling/models/spectral.py:299]: Done.

**Comment by adonath** [comment]
Thanks a lor @fabiopintore! I noties one more thing: you accidentally committed the `gammapy-extra` as a sub-repository. Please remove. In general it is not advisable to checkout a repository into another one...

**Comment by fabiopintore** [comment]
Oh right, my mistake. I've removed it.

**Comment by adonath** [comment]
Thanks @fabiopintore! I have seen one last test fail here https://github.com/gammapy/gammapy/pull/3402/checks?check_run_id=2952829917#step:9:1935

**Comment by adonath** [comment]
Thanks @fabiopintore!

**Pull request merged**